src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1551,7 +1551,7 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
         if (tryParseMethodCall is null)
         {
             var typeName = TypeNameHelper.GetTypeDisplayName(targetParseType, fullName: false);
-            throw new InvalidOperationException($"No public static bool {typeName}.TryParse(string, out {typeName}) method found for {parameter.Name}.");
+            throw new InvalidOperationException($"{parameter.Name} must have a valid TryParse method to support complex binding. No public static bool {typeName}.TryParse(string, out {typeName}) method found for {parameter.Name}.");
```

Comment:
```suggestion
            throw new InvalidOperationException($"{parameter.Name} must have a valid TryParse method to support converting from a string. No public static bool {typeName}.TryParse(string, out {typeName}) method found for {parameter.Name}.");
```

I think TypeConverter/TryParse binding would be considered "simple binding" in MVC parlance. https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#model-binding-simple-and-complex-types

Reply:
Ah, fair point. "converting from a string" is also a clearer description of what is going on under the hood.